### PR TITLE
Add controller argument to SubmenuButton

### LINKED ISSUE #124988: Cannot open a Submenu from another widget
The MenuBar widget takes a MenuController to override the default one.
SubmenuButton does not.

Without that, it does not appear to open a submenu programmatically.

## Use case

On web on Mac, the menu cannot be accessed with the Alt key because Mac does not work this way.
Ctrl-F2 is the shortcut used on mac to access the system menu, but that does not give us access to the menu being rendered inside the Chrome page.

Google Docs uses the key combination Option+/ to open the Docs menu.
A menu needs to be reachable with the keyboard for accessibility issues.
This could be done through a separate shortcut action if the MenuController could be overridden on the SubmenuButton.
The MenuController for the MenuBar is useless as the top menu bar is always considered open.

## Proposal

add a MenuController? parameter to SubmenuButton, and use it instead of the one created in the state object.
In addition, and that's optional, SubmenuButton should wrap MenuAnchor.onOpen to set the focus to the focus node an that next frame, similar to what is done in "toggleShowMenu", so that the external widget can simply call "open" on the controller to simulate the menu element being clicked.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds an optional argument to the `SubmenuButton` that allows the creator to supply a `MenuController` for controlling the menu.

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/124988

## Tests
 - Added tests for new argument.

### CODE REVIEW

**Review by saintmac-google — COMMENTED** [review]
the listener on the focus node is not being removed if the focusNode is provided through the widget.
line 1793 should probably say _buttonFocusNode.removeListener(_handleFocusChange)

File: dev/snippets/config/skeletons/sample.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <div class="copyable-container">
     <button class="copy-button-overlay copy-button" title="Copy to clipboard"
       onclick="copyTextToClipboard(longSnippet{{serial}});">
-      <i class="material-icons copy-image">assignment</i>
+      <i class="material-icons copy-image">content_copy</i>
```
Comment (QuncCccccc) [inline dev/snippets/config/skeletons/sample.html:17]: Seems this and the change below is not related to this PR. And looks like this change has been merged to master:)

**Review by QuncCccccc — COMMENTED** [review]
Looks like one unrelated commit is accidentally added to this PR. Otherwise, LGTM!

File: dev/snippets/config/skeletons/sample.html (hunk 1 above)
Comment (gspencergoog) [reply dev/snippets/config/skeletons/sample.html:17]: Oh!  Thanks, yes, that's not related.

**Review by QuncCccccc — APPROVED** [review]
LGTM!
